opentelemetry-config/src/model/metrics/reader.rs

```diff
@@ -0,0 +1,171 @@
+//! Metrics Reader Configuration models
+//!
+//! This module defines the configuration structures and factory traits
+//! for Metrics readers used in OpenTelemetry SDKs.
+
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+/// Metrics reader configuration
+#[derive(Debug)]
+pub enum Reader {
+    Periodic(Periodic),
+    Pull(Pull),
+}
+
+/// Custom deserialization for Reader enum to handle different reader types
+impl<'de> Deserialize<'de> for Reader {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let map: HashMap<String, serde_yaml::Value> = HashMap::deserialize(deserializer)?;
+
+        if let Some((key, value)) = map.into_iter().next() {
+            match key.as_str() {
+                "periodic" => {
+                    let variant: Periodic =
+                        serde_yaml::from_value(value).map_err(serde::de::Error::custom)?;
+                    Ok(Reader::Periodic(variant))
+                }
+                "pull" => {
+                    let variant: Pull =
+                        serde_yaml::from_value(value).map_err(serde::de::Error::custom)?;
+                    Ok(Reader::Pull(variant))
+                }
+                _ => Err(serde::de::Error::unknown_variant(
+                    &key,
+                    &["periodic", "pull"],
+                )),
+            }
+        } else {
+            Err(serde::de::Error::custom("Empty map"))
+        }
+    }
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct Periodic {
+    pub exporter: Option<PeriodicExporter>,
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct PeriodicExporter {
```

Comment:
It seems `deny_unknown_fields` will block adding any custom periodic exporters. Something like this won't work:
```yaml
          metrics:
            readers:
              - periodic:
                  exporter:
                    genevaexporter:
                      endpoint: "http://localhost:14268"
                      service_name: "my-service"
```